pkg/operator/ceph/object/bucket/controller.go

```diff
@@ -159,17 +152,10 @@ func (r *ReconcileBucket) reconcile(request reconcile.Request) (reconcile.Result
 		r.opConfig.Parameters = opConfig.Data
 	}
 
-	// Populate clusterInfo during each reconcile
-	clusterInfo, _, _, err := opcontroller.LoadClusterInfo(r.context, r.opManagerContext, cephCluster.Namespace, &cephCluster.Spec)
-	if err != nil {
-		// This avoids a requeue with exponential backoff and allows the controller to reconcile
-		// more quickly when the cluster is ready.
-		if errors.Is(err, opcontroller.ClusterInfoNoClusterNoSecret) {
-			return opcontroller.WaitForRequeueIfOperatorNotInitialized, nil
-		}
-		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to populate cluster info")
+	clusterInfo := &cephclient.ClusterInfo{
+		Namespace: cephObjectStore.Namespace,
+		Context:   r.opManagerContext,
 	}
-	r.clusterInfo = clusterInfo
 
 	// Start the object bucket provisioner
 	bucketProvisioner := NewProvisioner(r.context, clusterInfo)
```

Comment:
I think I finally realize why this code originally used CephCluster instead of CephObjectStore. The code is starting a new bucket controller for each namespace where a CephCluster is present, effectively. 

If we continue to create a new OBC controller for each namespace, we need to be careful not to create a new OBC controller for every CephObjectStore, because some users like those in situations @arttor is looking at currently could have several object stores. We don't want multiple OBC controllers running per namespace, which will eat up unnecessary system resources.

Thinking more about this, I have a theory that Rook really only needs to run one single OBC controller for the whole K8s cluster (same idea as CSI drivers). I'm not even sure we should have a separate OBC controller per namespace. That seems like it also eats up unnecessary resources to me. IIRC, each OBC StorageClass has to define the namespace and name of the CephObjectStore to use, so the identifying information should be already recorded when an OBC Provision/Grant request comes through.

If we were to run a single OBC controller per Rook operator, I think that could also make it easier to restart the OBC controller when CephObjectStore endpoints change, so investigating this could have benefits for other tasks we've had on our plate for a while. I'm not suggesting that we do that work here though; just a note for the future.

@parth-gr would you want to experiment with what it would be like to change the OBC controller to a one-per-operator approach? I think that could be a better overall change, but it could end up having ripple effects that make the change large and challenging. Getting an initial idea of what the scale and kinds of changes needed could be good to help us decide if it's worth pursuing more. LMK what you think

Reply:
>If we were to run a single OBC controller per Rook operator

We can still so it by `ROOK_OBC_WATCH_OPERATOR_NAMESPACE:  false` 
the reason to watch per cephcluster namespace is, it should watch for certain namespaces instead of all the namespace.

And we would be having per cephobjectstore per cephcluster, so that would be fine, and this will also fix the issue of reconciling if `CephObjectStore endpoints change`

Reply:
As discussed in the call, will limit the number of go-routines to number of cephcluster namespaces 

```
	go func() {
		err = bucketController.RunWithContext(r.opManagerContext)
```